app/controllers/admin/cms/transcripts_controller.rb

```diff
@@ -30,12 +30,35 @@ def update
     end
   end
 
+  def speaker_search
+    speakers = Speaker.where("LOWER(name) LIKE ?", "%#{params[:q].downcase}%").map {|s|
+      {id: s.id, label: s.name, value: s.name}
+    }
+    render json: speakers
+  end
+
   private
 
   def set_transcript
     @transcript = Transcript.find_by(uid: params[:id])
   end
 
+  def set_speakers(transcript)
+    # remove the exist speakers associated with the transcript
+    transcript.transcript_speakers.destroy_all
+
+    # replace the speakers with the edit and new form values
+    transcript_params[:speakers].split(';').reject { |c| c.blank? }.map do |name|
```

Comment:
don't think you need to use `map`, since you're not using the value returned. `each` will do what you need